src/com/hiltonrobotics/steamworksbot/commands/AutoCommand.java

```diff
@@ -1,135 +1,41 @@
 package com.hiltonrobotics.steamworksbot.commands;
 
-import com.hiltonrobotics.steamworksbot.OI;
-
 import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.CommandGroup;
 
-public class AutoCommand extends Command {
+public class AutoCommand extends CommandGroup {
 	public static double timestamp = 0;
 	
 	/*
 	 * place (0 = left, 1 = center, 2 = right): placement position out of 3 alliance robots
 	 * isOursRight (true = right, false = left): which side of the nearest scale we own
 	 */
-	private int place;
-	private boolean isOursRight;
 	
 	public AutoCommand(int placeIn, boolean isOursRightIn) {
-		place = placeIn;
-		isOursRight = isOursRightIn;
-	}
-	
-	private Command subcommand = null;
-	private int state = -1;
-	public void execute() {
-		if (subcommand != null) {
-			if (subcommand.isRunning()) {
-				return;
-			} else {
-				System.out.println("Finished " + subcommand.getName() + " in ~" + subcommand.timeSinceInitialized() + " seconds");
-				subcommand = null;
-			}
-		}
-		++state;
+		int place = placeIn;
+		boolean isOursRight = isOursRightIn;
 		int offset = (isOursRight ? 2 : 0) - place;
-		switch (state) {
-			case 0:
-				if (offset == 0) {
-					subcommand = new MoveCommand(30);
-					state = 4;
-				} else {
-					subcommand = new MoveCommand(8);
-				}
-				break;
-			case 1:
-				subcommand = new TurnCommand((offset > 0) ? -90 : 90, 0.5);
-				break;
-			case 2:
-				subcommand = new MoveCommand(Math.abs(offset) * 26);
-				break;
-			case 3:
-				subcommand = new TurnCommand((offset > 0) ? 90 : -90);
-				break;
-			case 4:
-				subcommand = new MoveCommand(22);
-				break;
-			case 5:
-				subcommand = new TurnCommand(isOursRight ? -90 : 90);
-				break;
-			case 6:
-				subcommand = new ArmExtremeCommand(true);
-				break;
-			case 7:
-				subcommand = new ShoveCommand(1.5, -0.4);//MoveCommand(-3);
-				break;
-			case 8:
-				subcommand = new ClawCommand(true);
-				break;
-			case 9:
-				subcommand = new MoveCommand(4);
-				break;
-			case 10:
-				subcommand = new ClawCommand(false);
-				break;
-			/*case 11:
-				subcommand = new ArmExtremeCommand(false);
-				break;*/
-			default:
-				this.cancel();
-				System.out.println("Ending autonomous...");
-				return;
-		}
-		subcommand.start();
-		
-		/*
-		if (timestamp < 0.5) {
-			OI.rightMotor.setSpeed(0.5);
-			OI.leftMotor.setSpeed(0.5);
+		if (offset == 0) {
```

Comment:
Please add a simple comment for this, e.g. "the robot is on the same side as the scale"

Reply:
Added an offset command

Reply:
Describing the meaning of offset